### ISSUE TITLE
`require-throws`: false positive in `async` function

### ISSUE THREAD

**Issue by regseb** [gajus/eslint-plugin-jsdoc#722]
## Expected behavior
Do not report [`require-throws`](https://github.com/gajus/eslint-plugin-jsdoc#eslint-plugin-jsdoc-rules-require-throws) when a `throw` is added in an `async` function (because the function returns a rejected promise).

## Actual behavior
If an `async` function contains a `throw`, ESLint report `require-throws` rule:

```
/home/regseb/testcase/index.js
  1:1  error  Missing JSDoc @throws declaration  jsdoc/require-throws

✖ 1 problem (1 error, 0 warnings)
```

## ESLint Config

```JSON
{
  "plugins": ["jsdoc"],
  "parserOptions": {
    "ecmaVersion": 2017
  },
  "rules": {
    "jsdoc/require-throws": 2
  }
}
```

## ESLint sample

```js
/**
 */
async function foo() {
  throw Error("bar");
}
```

## Environment

- Node version: _14.16.1_
- ESLint version: _7.25.0_
- `eslint-plugin-jsdoc` version: _33.0.0_

**Comment by brettz9** [comment]
A function with the `async` keyword will never be able to have `throws`, but I think it is meaningful to have a means of documenting the types that are rejected.

FWIW, TypeScript assumes "any" type for the rejected value, even while allowing the resolved value to be specified. Barring such a convention for specifying the rejected value, I think we need to allow for a means of requiring that the rejected value be documented.

There could admittedly be ambiguity in functions which are only async in having an `@async` tag, but which are regular functions which may throw synchronously as well as perform some async behavior (which may reject), but without a convention, I think it is better for this function to require the error types be documented, and consumers will just have to know that "throws" here refers to the rejection type given that the function is `async`.

We could potentially add config to prevent this, however.
